pkg/project/admission/requestlimit/admission.go

```diff
@@ -0,0 +1,149 @@
+package requestlimit
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"reflect"
+
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/labels"
+	errutil "k8s.io/kubernetes/pkg/util/errors"
+
+	"github.com/openshift/origin/pkg/client"
+	oadmission "github.com/openshift/origin/pkg/cmd/server/admission"
+	configlatest "github.com/openshift/origin/pkg/cmd/server/api/latest"
+	projectapi "github.com/openshift/origin/pkg/project/api"
+	projectcache "github.com/openshift/origin/pkg/project/cache"
+)
+
+func init() {
+	admission.RegisterPlugin("ProjectRequestLimit", func(client kclient.Interface, config io.Reader) (admission.Interface, error) {
+		pluginConfig, err := readConfig(config)
+		if err != nil {
+			return nil, err
+		}
+		return NewProjectRequestLimit(pluginConfig)
+	})
+}
+
+func readConfig(reader io.Reader) (*ProjectRequestLimitConfig, error) {
+	if reader == nil || reflect.ValueOf(reader).IsNil() {
+		return &ProjectRequestLimitConfig{}, nil
+	}
+
+	configBytes, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+	config := &ProjectRequestLimitConfig{}
+	err = configlatest.ReadYAML(configBytes, config)
+	if err != nil {
+		return nil, err
+	}
+	errs := ValidateProjectRequestLimitConfig(config)
+	if len(errs) > 0 {
+		return nil, errutil.NewAggregate(errs)
+	}
+	return config, nil
+}
+
+type projectRequestLimit struct {
+	*admission.Handler
+	client client.Interface
+	config *ProjectRequestLimitConfig
+	cache  *projectcache.ProjectCache
+}
+
+// ensure that the required Openshift admission interfaces are implemented
```

Comment:
I like it too :) Allowed us to remove the embarrassing hack that merged the openshift and kube clients. Thank you for suggesting it.